pages/[resourceType]/[id]/update.tsx

```diff
@@ -0,0 +1,133 @@
+ * appears and the user is redirected to the resource's home. Otherwise, an error notification appears.
+ * @returns React node with a ResourceCodeEditor component, a "submit" Button, and a back button
+ */
+const UpdateResourcePage = () => {
+  const router = useRouter();
+  const { resourceType, id } = router.query;
+  const [codeEditorContents, setCodeEditorContents] = useState("");
+  const [hasLintError, setHasLintError] = useState(true);
+
+  const [pageBody, setPageBody] = useState("");
+  useEffect(() => {
+    if (resourceType && id) {
+      //fetch the resource JSON content from the test server based on resource and id from url
+      fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}/${id}`)
+        .then((data) => {
+          return data.json();
+        })
+        .then((resourcePageBody) => {
+          setPageBody(JSON.stringify(resourcePageBody, null, 2));
+        });
+    }
+  }, [resourceType, id]);
+
+  return (
+    <div style={{ paddingLeft: "15px", paddingRight: "15px" }}>
+      <BackButton />
+      <Stack spacing="xs">
+        <Center>
+          <h2 style={{ color: textGray, marginTop: "2px" }}> Update Resource</h2>
+        </Center>
+        <Center>
+          <Text size="md" color={textGray}>
+            Edit the FHIR resource body with valid JSON. Click Update Resource to update.
+          </Text>
+        </Center>
+      </Stack>
+      <div style={{ paddingTop: "20px", paddingBottom: "20px" }}>
+        <ResourceCodeEditor
+          initialValue={pageBody}
+          onUpdate={(submittedVal) => setCodeEditorContents(submittedVal)}
+          onValidate={(hasError) => setHasLintError(hasError)}
+        />
+      </div>
+      <Center>
+        <Button
+          disabled={hasLintError}
+          onClick={editorSubmitHandler}
+          color="cyan"
+          variant="filled"
+          size="lg"
+        >
+          Update Resource
+        </Button>
+      </Center>
+    </div>
+  );
+
+  //called when submit button is clicked. Handles PUT request and response
+  function editorSubmitHandler() {
+    let customMessage: NotificationProps["message"] = <div>Problem connecting to server</div>;
+    let notifProps: NotificationProps = {
+      message: customMessage,
+      color: "red",
+      icon: <X size={18} />,
+      autoClose: false,
+    };
+
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}/${id}`, {
+      method: "PUT",
+      body: codeEditorContents,
+      headers: {
+        "Content-Type": "application/json+fhir",
+      },
+    })
+      .then((response) => {
+        if (response.status === 201 || response.status === 200) {
+          customMessage = (
+            <>
+              <Text>Resource successfully updated!&nbsp;</Text>
+            </>
+          );
+          notifProps = {
+            ...notifProps,
+            color: "green",
+            icon: <Check size={18} />,
+          };
+
+          //redirects user to page with the resource's body
```

Comment:
This is more of a suggestion and it probably has more to do with the back button functionality, but when a user updates the resource and is redirected back to the page displaying the JSON body (which is great!!) The thing I am curious about is then when the user clicks the back button, they are back to the code editor for that resource. I think it would make more sense for it to go back to the list of resources, but if you wanted more of an _actual_ back button (meaning literally takes you BACK to where you were) then disregard this :)